deployment/common/changeset/set_config_mcms_test.go

```diff
@@ -0,0 +1,327 @@
+package changeset
+
+import (
+	"context"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	chain_selectors "github.com/smartcontractkit/chain-selectors"
+
+	"github.com/smartcontractkit/chainlink/deployment/common/proposalutils"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap/zapcore"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+
+	"github.com/smartcontractkit/chainlink/deployment/common/types"
+	"github.com/smartcontractkit/chainlink/deployment/environment/memory"
+)
+
+// setupSetConfigTestEnv deploys all required contracts for the setConfig MCMS contract call.
+func setupSetConfigTestEnv(t *testing.T) deployment.Environment {
+
+	lggr := logger.TestLogger(t)
+	cfg := memory.MemoryEnvironmentConfig{
+		Nodes:  1,
+		Chains: 2,
+	}
+	env := memory.NewMemoryEnvironment(t, lggr, zapcore.DebugLevel, cfg)
+	chainSelector := env.AllChainSelectors()[0]
+	config := proposalutils.SingleGroupMCMS(t)
+
+	// Deploy MCMS and Timelock
+	env, err := ApplyChangesets(t, env, nil, []ChangesetApplication{
+		{
+			Changeset: WrapChangeSet(DeployLinkToken),
+			Config:    []uint64{chainSelector},
+		},
+		{
+			Changeset: WrapChangeSet(DeployMCMSWithTimelock),
+			Config: map[uint64]types.MCMSWithTimelockConfig{
+				chainSelector: {
+					Canceller:        config,
+					Bypasser:         config,
+					Proposer:         config,
+					TimelockMinDelay: big.NewInt(0),
+				},
+			},
+		},
+	})
+	require.NoError(t, err)
+	return env
+}
+
+// TestSetConfigMCMS tests the SetConfigMCMS changeset by calling SetConfig and checking the config values.
+func TestSetConfigMCMS(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
```

Comment:
use tests.Context(t) from the common repo

Reply:
added here https://github.com/smartcontractkit/chainlink/pull/15684/commits/f12253f6984727db67444588bb4721511ed4acff

Reply:
added here https://github.com/smartcontractkit/chainlink/pull/15684/commits/f12253f6984727db67444588bb4721511ed4acff

Reply:
added here https://github.com/smartcontractkit/chainlink/pull/15684/commits/f12253f6984727db67444588bb4721511ed4acff